compound.hh

```diff
@@ -156,6 +156,7 @@ public:
             read_current();
         }
         iterator(end_iterator_tag, const bytes_view& v) : _v(nullptr, 0) {}
+        iterator() : iterator(end_iterator_tag()) {}
```

Comment:
where is `iterator(end_iterator_tag)` defined? Is it in std::iterator?

Reply:
It is defined one line about (line 158).

Reply:
158 is `iterator(end_iterator_tag, const bytes_view& v)`

Reply:
That's different than `iterator(end_iterator_tag)`, no?

Reply:
yes, it is.

Reply:
I'm guessing it's broken, but the compiler deferred instantiation because this is a template, and the constructor body was not needed anywhere (we only need the definition for the concept check). I'll fix it and check the other patch for the same problem.

Reply:
The other commit is fine.

Reply:
(moral: avoid snarky comments, they turn around and bite you)